Update nextcloud to version v31

nmfretz opened the pull request:
🤖 This is an automated pull request to update nextcloud to version v30.0.6. This PR must be reviewed and tested before merging.

**Auto-generated update considerations**:
No bullet points to report - this release contains only minor fixes without any changes that would affect Docker deployments, database migrations, or require user intervention.

[Original GitHub release notes](https://github.com/nextcloud/server/releases/tag/v30.0.6)

nmfretz commented:
🤖 This is an automated summary of installation tests for nextcloud:

- ✅ App successfully installed in umbrel-dev instance
- ✅ App successfully listed in umbrel.yaml
- ✅ App UI successfully returns a 200 status code
- 📸 Screenshot of the app:
  ![App Screenshot](https://raw.githubusercontent.com/nmfretz/app-testing-screenshots/main/screenshots/1739772277414_nextcloud_2025-02-17T06-04-37-120Z.png)

**This PR must still be reviewed and tested before merging.**

nmfretz commented:
This PR also upgrades the mariadb container to 3.6.12 (3.6 is still the tested and recommended version of mariadb version according to Nextcloud). This db version is compatible with all Nextcloud version updates that are on the umbrelOS app store. Addresses https://github.com/getumbrel/umbrel-apps/pull/1547#issuecomment-2385247354

@al-lac @Sharknoon - I would love some additional testing of this one due to the db upgrade.

nmfretz commented:
Update from 30.0.5 to 30.0.6 failed for me... looking into it:

<img width="869" alt="image" src="https://github.com/user-attachments/assets/8738d3b0-6f54-47e5-ac37-7144820e20cd" />

al-lac commented on nextcloud/docker-compose.yml:
```diff
@@ -19,6 +19,7 @@ services:
       - MYSQL_PASSWORD=moneyprintergobrrr
       - MYSQL_DATABASE=nextcloud
       - MYSQL_USER=nextcloud
+      - MARIADB_AUTO_UPGRADE=1
```
I would add a healthcheck here to make sure the db is available before starting nextcloud.
```suggestion
      - MARIADB_AUTO_UPGRADE=1
    healthcheck:
      test: ["CMD", "healthcheck.sh", "--connect", "--innodb_initialized"]
      start_period: 10s
      interval: 10s
      timeout: 5s
      retries: 3
```

al-lac commented on nextcloud/docker-compose.yml:
```diff
@@ -51,7 +52,7 @@ services:
       - redis
```
Cannot mark the whole depends_on here for some reason. But that is how it should look like.
```suggestion
    depends_on:
      db:
        condition: service_healthy
      redis:
        condition: service_started
```

al-lac commented on nextcloud/docker-compose.yml:
```diff
@@ -61,4 +62,4 @@ services:
     entrypoint: /cron.sh
     depends_on:
       - db
```
```suggestion
    depends_on:
      db:
        condition: service_healthy
      redis:
        condition: service_started
```

al-lac reviewed (changes requested):
@nmfretz @sharknoon for me the update failed with this error multiple times on my Umbrel Home:
```
nextcloud_web_1        | Configuring Redis as session handler
nextcloud_web_1        | Initializing nextcloud 30.0.6.2 ...
nextcloud_web_1        | Upgrading nextcloud from 30.0.5.1 ...
nextcloud_web_1        | => Searching for scripts (*.sh) to run, located in the folder: /docker-entrypoint-hooks.d/pre-upgrade
nextcloud_web_1        | An unhandled exception has been thrown:
nextcloud_web_1        | Doctrine\DBAL\Exception: Failed to connect to the database: An exception occurred in the driver: SQLSTATE[HY000] [2002] Connection refused in /var/www/html/lib/private/DB/Connection.php:233
nextcloud_web_1        | Stack trace:
```

Adding the db health check solved this issue for me, the update went through just fine afterwards.

nmfretz commented:
> @nmfretz I have tested this update and it works fine on my system. I have changed some settings and uploaded some photos in the old version of Nextcloud, updated it and it was still there :)

Thanks @Sharknoon! I've now tried various updates from past versions even without the new health-checks and they all worked. I suspect this was a random failure that non-AIO Nextcloud is natorious for. Looking at forums and their github it looks like updating to 30.0.6 has quite a lot of reported errors from users.

> btw, I just found out, that Nextcloud is being developed in Stuttgart, where I live. One of the example photos was our library :)

wooahhh that is an amazing looking library!

> EDIT: regarding your failed update, I have not tested the browser update, only the update via the umbrel-cli, which basically calls umbreld client apps.update.mutate --appId ${appId}


@al-lac thanks for the healthcheck updates, I've committed them.

> Adding the db health check solved this issue for me, the update went through just fine afterwards.

Unfortunately it doesn't fix my particular issue. I think I'm in a funky state that I'm going to have to get my hands dirty manually running commands and editing files to get out of.

@sharknoon @al-lac I vote we update to v31.0 which just came out 2 days ago. We'll skip 30.0.6 which seems to have some issues. Hopefully 31 is more stable.

nmfretz commented:
@sharknoon @al-lac - updating to v31 fixed my borked Nextcloud! Whether it was due to the update triggering the hook properly somehow, or a bugfix in 31 I'm not sure.

al-lac commented:
@nmfretz update to v31 worked fine for me! Maybe the release notes can be updated, this is what I got on the Web UI in the whats new section:

> 🔒 End-to-end encryption in calls and web
> ⚡ Performance improvements across all our products
> ⚙️ Thousands of tweaks for improved UX and stability
> ✨ Your own new personal assistant
> 🔄 Seamless file conversions
> 👥 Smarter team sharing with Team folders
> 💬 Schedule meetings directly in Talk
> 📧 Mail you love again: translations, summaries, and mentions

nmfretz commented:
@al-lac thanks for those release notes! That's a great way to grab them for future updates.

Updated:
<img width="845" alt="image" src="https://github.com/user-attachments/assets/d1011012-d988-49f0-8fa6-d99f4dce3727" />
